Build failure when building for LPC11U35_401 with ARMC6

Building the example for LPC11U35_401 with toolchain ARMC6 results in the following build error:
~~~
[Warning] @0,0: L3912W: Option 'legacyalign' is deprecated.
[Warning] @0,0: L6320W: Ignoring --keep command. Cannot find argument 'os_cb_sections'.
[Error] @0,0: L6218E: Undefined symbol _scanf_mbtowc (referred from ../BUILD/LPC11U35_401/ARMC6/mbed-os/platform/ATCmdParser.o).
[Error] @0,0: L6218E: Undefined symbol us_ticker_irq_handler (referred from ../BUILD/LPC11U35_401/ARMC6/mbed-os/targets/TARGET_NXP/TARGET_LPC11UXX/us_ticker.o).
Warning: L3912W: Option 'legacyalign' is deprecated.
Warning: L6320W: Ignoring --keep command. Cannot find argument 'os_cb_sections'.
Error: L6218E: Undefined symbol _scanf_mbtowc (referred from ../BUILD/LPC11U35_401/ARMC6/mbed-os/platform/ATCmdParser.o).
Error: L6218E: Undefined symbol us_ticker_irq_handler (referred from ../BUILD/LPC11U35_401/ARMC6/mbed-os/targets/TARGET_NXP/TARGET_LPC11UXX/us_ticker.o).
Finished: 0 information, 2 warning and 2 error messages.
[ERROR] Warning: L3912W: Option 'legacyalign' is deprecated.
Warning: L6320W: Ignoring --keep command. Cannot find argument 'os_cb_sections'.
Error: L6218E: Undefined symbol _scanf_mbtowc (referred from ../BUILD/LPC11U35_401/ARMC6/mbed-os/platform/ATCmdParser.o).
Error: L6218E: Undefined symbol us_ticker_irq_handler (referred from ../BUILD/LPC11U35_401/ARMC6/mbed-os/targets/TARGET_NXP/TARGET_LPC11UXX/us_ticker.o).
Finished: 0 information, 2 warning and 2 error messages.

[mbed] ERROR: "/usr/bin/python" returned error.
       Code: 1
       Path: "/home/roneld01/work/mbed-os-example-blinky-baremetal/mbed-os"
       Command: "/usr/bin/python -u /home/roneld01/work/mbed-os-example-blinky-baremetal/mbed-os/tools/make.py -t ARMC6 -m LPC11U35_401 --source .. --build ../BUILD/LPC11U35_401/ARMC6"
~~~
 
I believe the `Undefined symbol _scanf_mbtowc` error has been fixed in https://github.com/ARMmbed/mbed-os/pull/10704 and the `mbed-os.lib` requires updating to latest versoin, once next Mbed OS release is published.
 I believe  error `Undefined symbol us_ticker_irq_handler` is because the [targets.json](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L360) doesn't have device `US_TICKER` for `LPC11U35_401` . To resolve this, either add `US_TICKER` DEVICE in ``targets.json` or add `#if  DEVICE_USTICKER` in the targets' [us_ticker.c](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_NXP/TARGET_LPC11UXX/us_ticker.c) file

Note that same errors happen with `ARM` toolchain which is mentioned as supported toolchain

This was reported in the [forum](https://os.mbed.com/forum/mbed/topic/36091/)

@0xc0170 @RonEld 

I just quick test for this platform.  My test branch is here:
https://github.com/toyowata/mbed/tree/test_lpc11u35

I did...
* Add `USTICKER` for the target
* Add `#if DEVICE_USTICKER` macro in the us_ticker.c

```
(mbed-venv) 16:37:35 mbed-os-example-blinky-baremetal $ mbed compile -m lpc11u35_401 -t arm
[mbed] Working path "/Users/toywat01/test/nxp/mbed-os-example-blinky-baremetal" (program)
Building project mbed-os-example-blinky-baremetal (LPC11U35_401, ARMC6)

# snip

Link: mbed-os-example-blinky-baremetal
[Warning] @0,0: L3912W: Option 'legacyalign' is deprecated.
[Warning] @0,0: L6320W: Ignoring --keep command. Cannot find argument 'os_cb_sections'.
[Error] @0,0: L6218E: Undefined symbol _scanf_mbtowc (referred from BUILD/LPC11U35_401/ARM/mbed-os/platform/source/ATCmdParser.o).
Warning: L3912W: Option 'legacyalign' is deprecated.
Warning: L6320W: Ignoring --keep command. Cannot find argument 'os_cb_sections'.
Error: L6218E: Undefined symbol _scanf_mbtowc (referred from BUILD/LPC11U35_401/ARM/mbed-os/platform/source/ATCmdParser.o).
Finished: 0 information, 2 warning and 1 error messages.
[ERROR] Warning: L3912W: Option 'legacyalign' is deprecated.
Warning: L6320W: Ignoring --keep command. Cannot find argument 'os_cb_sections'.
Error: L6218E: Undefined symbol _scanf_mbtowc (referred from BUILD/LPC11U35_401/ARM/mbed-os/platform/source/ATCmdParser.o).
Finished: 0 information, 2 warning and 1 error messages.

[mbed] ERROR: "/Users/toywat01/mbed-venv/bin/python" returned error.
       Code: 1
       Path: "/Users/toywat01/test/nxp/mbed-os-example-blinky-baremetal"
       Command: "/Users/toywat01/mbed-venv/bin/python -u /Users/toywat01/test/nxp/mbed-os-example-blinky-baremetal/mbed-os/tools/make.py -t arm -m lpc11u35_401 --source . --build ./BUILD/LPC11U35_401/ARM"
       Tip: You could retry the last command with "-v" flag for verbose output
---
```

I still got link erros, so I changed:
* Change default_toolchain from `uARM` to `ARM`

Now, I got no link errors.  Then, I copied the mbed-os-example-blinky-baremetal binary to the target and got runtime error below:

```

++ MbedOS Fault Handler ++

FaultType: HardFault

Context:
R0   : 00000013
R1   : 00003355
R2   : 00000002
R3   : 00000002
R4   : 00000001
R5   : 40018004
R6   : 00000000
R7   : 0000399F
R8   : 19494B1A
R9   : 10000140
R10  : 000039C0
R11  : 000039C0
R12  : 00000000
SP   : 10001F88
LR   : 00003335
PC   : 00001F80
xPSR : 21000000
PSP  : 7821E644
MSP  : 10001F68
CPUID: 410CC200
Mode : Thread
Priv : Privileged
Stack: MSP

-- MbedOS Fault Handler --



++ MbedOS Error Info ++
Error Status: 0x80FF013D Code: 317 Module: 255
Error Message: Fault exception
Location: 0x1F80
Error Value: 0x10000180
For more info, visit: https://mbed.com/s/error?error=0x80FF013D&tgt=LPC11U35_401
-- MbedOS Error Info --
```

Note that `-t gcc_arm` binary doesn't crash and works as expected (LED blinks every 500ms) on the target.

The LPC11U35_401 is Mbed 2 only target.  There should be done proper porting/testing for Mbed OS 5 even baremetal profile?

--- a/targets/TARGET_NXP/TARGET_LPC11UXX/device/cmsis_nvic.c
+++ b/targets/TARGET_NXP/TARGET_LPC11UXX/device/cmsis_nvic.c
@@ -60,7 +60,8 @@ void NVIC_SetVector(IRQn_Type IRQn, uint32_t vector) {
     
     // Copy and switch to dynamic vectors if first time called
     if((LPC_SYSCON->SYSMEMREMAP & 0x3) != 0x1) {     
-      uint32_t *old_vectors = (uint32_t *)0;         // FLASH vectors are at 0x0
+      // Add volatile qualifier to avoid armclang aggressive optimization
+      volatile uint32_t *old_vectors = (uint32_t *)0;         // FLASH vectors are at 0x0
       for(i = 0; i < NVIC_NUM_VECTORS; i++) {    
             vectors[i] = old_vectors[i];
         }

--- a/targets/TARGET_NXP/TARGET_LPC11UXX/us_ticker.c
+++ b/targets/TARGET_NXP/TARGET_LPC11UXX/us_ticker.c
@@ -17,6 +17,8 @@
 #include "us_ticker_api.h"
 #include "PeripheralNames.h"
 
+#if DEVICE_USTICKER
+
 #define US_TICKER_TIMER          ((LPC_CTxxBx_Type *)LPC_CT32B1_BASE)
 #define US_TICKER_TIMER_IRQn     TIMER_32_1_IRQn
 
@@ -70,3 +72,5 @@ void us_ticker_free(void)
 {
 
 }
+
+#endif // DEVICE_USTICKER

--- a/targets/targets.json
+++ b/targets/targets.json
@@ -374,8 +374,13 @@
             "SERIAL",
             "SLEEP",
             "SPI",
-            "SPISLAVE"
+            "SPISLAVE",
+            "USTICKER"
         ],
+        "overrides": {
+            "tickless-from-us-ticker"   : true,
+            "boot-stack-size"           : "0x400"
+        },
         "default_lib": "small",
         "release_versions": ["2"],
         "device_name": "LPC11U35FBD48/401"

